### PR TITLE
[YUNIKORN-2491] Selects default queue value if one exists

### PR DESCRIPTION
**Pull request by dcoric**
Check how prometheus is used from YU
Pull logs into a separate ticket

Improvement on the Applications page

### What is this PR for?
This an improvement on the Applications page. The partition dropdown is preselected, but queue dropdown shows "select". With this improvement, if there is a `.default` queue, that one will be selected automatically. 

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### What is the Jira issue?
[YUNIKORN-2491](https://issues.apache.org/jira/browse/YUNIKORN-2491) - `Preselect default Queue on Applications page`

### CODE REVIEW

**Comment by chenyulin0719** [comment]
Hi @dcoric,
Thanks for the PR. Just leaving a comment in Jira, waiting to see if there are any feedbacks from others.

**Comment by craigcondit** [comment]
I don't think we should codify the default queue. It's only the default if users don't customize YuniKorn configuration at all, and is not guaranteed to even exist. I do see a need for a better experience here -- having to select the queue each time is cumbersome. Perhaps instead we could keep a little bit of client-side state that automatically selects the last used queue if it exists (and if not fallback to no queue selected). This could flow across all pages which have queue selections. This way a queue only needs to be selected once.

**Comment by dcoric** [comment]
It makes sense. I have added a comment to Jira. I will proceed with an update and remove "default" queue

**Comment by pbacsko** [comment]
ping @dcoric have you time to make those changes? Thanks.

**Comment by dcoric** [comment]
> ping @dcoric have you time to make those changes? Thanks.

Yes, sorry for the delay. I'm finishing up the updates.

File: src/app/components/queues-view/queues-view.component.ts [hunk 1]
```diff
@@ -226,4 +230,14 @@ export class QueuesViewComponent implements OnInit {
...
+
+  loadStoredPartition(defaultValue = ''): string {
+    const storedPartition = localStorage.getItem('selectedPartitionAndQueue');
+    if (storedPartition && storedPartition.indexOf(':') > 0) return storedPartition.split(':')[0];
+    return defaultValue;
+  }
+
+  storeSelectedPartition() {
+    localStorage.setItem('selectedPartitionAndQueue', this.partitionSelected + ':');
+  }
```
Comment (chenyulin0719) [inline src/app/components/queues-view/queues-view.component.ts:242]: Is it possible to move below localStorage functions into same module?  or move to utils? 
If it's ok we can create another Jira to refactor it.

1. (app-view) loadStoredPartition(defaultValue = '')
2. (app-view) storeQueueSelection()
3. (node-view) loadStoredPartition(defaultValue = '')
4. (node-view) storeSelectedPartition() {
5. (queue-view) loadStoredPartition(defaultValue = '')
6. (queue-view) storeSelectedPartition() {

File: src/app/components/apps-view/apps-view.component.ts [hunk 2]
```diff
@@ -161,14 +163,33 @@ export class AppsViewComponent implements OnInit {
...
+          this.setDefaultQueue(leafQueueList);
         } else {
           this.leafQueueList = [new DropdownItem('-- Select --', '')];
         }
       });
   }
 
+  setDefaultQueue(queueList: DropdownItem[]): void {
+    const storedPartitionAndQueue = localStorage.getItem('selectedPartitionAndQueue');
+    if (!storedPartitionAndQueue || storedPartitionAndQueue.indexOf(':') < 0) return;
```
Comment (chenyulin0719) [inline src/app/components/apps-view/apps-view.component.ts:176]: How about open queue selection if no data in local storage?

```
    if (!storedPartitionAndQueue || storedPartitionAndQueue.indexOf(':') < 0) {
      setTimeout(() => this.openQueueSelection(), 0);
      return;
    }
```

**Review by chenyulin0719 — COMMENTED** [review]
Overall LGTM.  @dcoric It's really cool!

File: src/app/components/queues-view/queues-view.component.ts (hunk 1 above)
Comment (dcoric) [reply src/app/components/queues-view/queues-view.component.ts:242]: It makes sense. I'm updating it with this ticket

**Review by chenyulin0719 — APPROVED** [review]
LGTM. Thanks for @dcoric's update.

Will merge it tomorrow.

**Pull request closed without merging**